clients/pkg/promtail/targets/file/tailer.go

```diff
@@ -217,6 +236,24 @@ func (t *tailer) isRunning() bool {
 	return t.running.Load()
 }
 
+func (t *tailer) convertToUTF8(text string) (string, error) {
+	level.Debug(t.logger).Log("msg", "Converting log encoding", "from", t.encoding, "to", "UTF8")
+
+	encoding, err := ianaindex.IANA.Encoding(t.encoding)
```

Comment:
You should create this only once ?

Reply:
done. I tested with 2 tailers and checked the source code and looks like we won't have concurrency problems reusing the decoder or the IANAEncoding struct, but do you mind checking it too?